src/overlays/actors/ovl_En_Takaraya/z_en_takaraya.c

```diff
@@ -28,61 +47,360 @@ ActorInit En_Takaraya_InitVars = {
+        } else if (this->skelAnime.animation == &object_bg_Anim_00A280) {
+            Animation_PlayLoop(&this->skelAnime, &object_bg_Anim_001384);
+        } else {
+            Animation_PlayLoop(&this->skelAnime, &object_bg_Anim_009890);
+        }
+    }
+    talkState = Message_GetState(&play->msgCtx);
+    if ((talkState == TEXT_STATE_CLOSING) || (talkState == TEXT_STATE_DONE)) {
+        if (this->actor.textId == 0x778) {
+            func_80ADF2D4(this);
+        } else {
+            CLEAR_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_WAIT);
+            CLEAR_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_TIME_PASSED);
+            func_80ADEDF8(this);
+        }
+    } else if ((talkState == TEXT_STATE_1) && (this->actor.textId != 0x778)) {
+        if (Message_ShouldAdvance(play)) {
+            Animation_MorphToPlayOnce(&this->skelAnime, &object_bg_Anim_00AD98, 5.0f);
+        }
+    } else if ((talkState == TEXT_STATE_CHOICE) && (Message_ShouldAdvance(play))) {
+        if (play->msgCtx.choiceIndex == 0) {
+            if (gSaveContext.save.saveInfo.playerData.rupees < play->msgCtx.unk1206C) {
+                this->actor.textId = 0x77B;
+                if (this->skelAnime.animation == &object_bg_Anim_009890) {
+                    Animation_MorphToPlayOnce(&this->skelAnime, &object_bg_Anim_000968, 5.0f);
+                }
+                Audio_PlaySfx(NA_SE_SY_ERROR);
+            } else {
+                Audio_PlaySfx_MessageDecide();
+                Rupees_ChangeBy(play->msgCtx.unk1206C * -1);
+                func_80ADEF74(this, play);
+                this->actor.textId = 0x778;
+                if (this->skelAnime.animation != &object_bg_Anim_009890) {
+                    Animation_MorphToLoop(&this->skelAnime, &object_bg_Anim_009890, 5.0f);
+                }
+            }
+        } else {
+            Audio_PlaySfx_MessageCancel();
+            this->actor.textId = D_80ADFB44[(void)0, gSaveContext.save.playerForm];
+            if (this->skelAnime.animation == &object_bg_Anim_009890) {
+                Animation_MorphToPlayOnce(&this->skelAnime, &object_bg_Anim_000968, 5.0f);
+            }
+        }
+        Message_ContinueTextbox(play, this->actor.textId);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/EnTakaraya_Init.s")
+void func_80ADF2D4(EnTakaraya* this) {
+    Actor_PlaySfx_FlaggedCentered2(&this->actor, NA_SE_SY_FOUND);
+    Audio_PlaySubBgm(NA_BGM_TIMED_MINI_GAME);
+    this->timer = 145;
+    SET_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_WAIT);
+    CLEAR_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_TIME_PASSED);
+    this->actionFunc = &func_80ADF338;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/EnTakaraya_Destroy.s")
+void func_80ADF338(EnTakaraya* this, PlayState* play) {
+    Actor* chest = play->actorCtx.actorLists[ACTORCAT_CHEST].first;
+    Actor* player = play->actorCtx.actorLists[ACTORCAT_PLAYER].first;
+    Vec3f eye;
+    Vec3f at;
+    f32 sp2C;
+    s16 CamId;
```

Comment:
```suggestion
    s16 subCamId;
```
You don't want to start local variables with a capital letter. Also, `subCamId` is the standard for this